enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyHandler.java

```diff
@@ -91,6 +93,25 @@ public void checkInRevokedMap(String tokenIdentifier, String[] splitToken) throw
         }
     }
 
+    /**
+     * Compares the audience claim with the deployment type of the API.
+     *
+     * @param tokenAudience list of intended audience of the API Key
+     * @param deploymentType Deployment type of the API
+     * @throws APISecurityException The comparison b/w the audience vs deployment type is invalid
+     */
+    public void compareAudience(List<String> tokenAudience, String deploymentType) throws APISecurityException {
+        if (tokenAudience != null && tokenAudience.size() > 0 &&
+                !StringUtils.isEmpty(deploymentType)) {
+            if (!tokenAudience.get(0).toLowerCase().contains(deploymentType.toLowerCase())) {
```

Comment:
Can we guarantee that the claim is sent as the 0th element, or should we iterate

Reply:
The aud claim was added as a task for this specific issue. So right now it's sent as the 0th and only element of aud array. WDYT ? @VirajSalaka

Reply:
ack